app/src/test/java/com/android/streetworkapp/model/park/ParkViewModelTest.kt

```diff
@@ -0,0 +1,190 @@
+package com.android.streetworkapp.model.park
+
+import com.android.streetworkapp.model.parklocation.ParkLocation
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.kotlin.whenever
+
+class ParkViewModelTest {
+
+  private lateinit var repository: ParkRepository
+  private lateinit var parkViewModel: ParkViewModel
+  private val testDispatcher = StandardTestDispatcher()
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @Before
+  fun setUp() {
+    Dispatchers.setMain(testDispatcher)
+    repository = mock()
+    parkViewModel = ParkViewModel(repository)
+  }
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @After
+  fun tearDown() {
+    Dispatchers.resetMain()
+  }
+
+  @Test
+  fun getNewPidCallsRepositoryAndReturnsPid() {
+    whenever(repository.getNewPid()).thenReturn("uniqueId")
+    val pid = parkViewModel.getNewPid()
+    assertEquals("uniqueId", pid)
+    verify(repository).getNewPid()
+  }
+
+  @Test
+  fun getParkByPidCallsRepositoryWithCorrectPid() = runTest {
+    val pid = "user123"
+    val park =
+        Park(
+            pid = pid,
+            name = "Sample Park",
+            location = ParkLocation(0.0, 0.0, "321"),
+            imageReference = "parks/sample.png",
+            rating = 4.0f,
+            nbrRating = 2,
+            capacity = 10,
+            occupancy = 5,
+            events = listOf("event1", "event2"))
+    whenever(repository.getParkByPid(pid)).thenReturn(park)
+    parkViewModel.getParkByPid(pid)
+    testDispatcher.scheduler.advanceUntilIdle()
+    verify(repository).getParkByPid(pid)
+  }
+
+  @Test
+  fun getParkByLocationIdCallsRepositoryWithCorrectLocationId() = runTest {
+    val locationId = "location123"
+    val park =
+        Park(
+            pid = "123",
+            name = "Sample Park",
+            location = ParkLocation(0.0, 0.0, locationId),
+            imageReference = "parks/sample.png",
+            rating = 4.0f,
+            nbrRating = 2,
+            capacity = 10,
+            occupancy = 5,
+            events = listOf("event1", "event2"))
+    whenever(repository.getParkByLocationId(locationId)).thenReturn(park)
+    parkViewModel.getParkByLocationId(locationId)
+    testDispatcher.scheduler.advanceUntilIdle()
+    verify(repository).getParkByLocationId(locationId)
```

Comment:
Same as the one below, you could verify the park you received

Reply:
Same as the other commentary answer.

Reply:
Same here